astropy/visualization/wcsaxes/tests/test_coordinate_helpers.py

```diff
@@ -205,3 +205,41 @@ def test_set_major_formatter():
 
     ax.coords[1].set_major_formatter("dd:mm:ss.s", show_decimal_unit=False)
     assert ax.coords[1].format_coord(4) == "4\xb000'00.0\""
+
+
+def test_set_visible():
+    fig = Figure()
+    canvas = FigureCanvasAgg(fig)
+    ax = WCSAxes(fig, [0.1, 0.1, 0.8, 0.8], wcs=WCS(MSX_HEADER))
+    fig.add_axes(ax)
+
+    coord = ax.coords[0]
+
+    coord.set_visible(False)
```

Comment:
Maybe also check that visibility behaves as expected _before_ it is explicitly set while we're at it ?